tidy3d/components/grid/grid.py

```diff
@@ -15,12 +15,15 @@
 # data type of one dimensional coordinate array.
 Coords1D = ArrayFloat1D
 
+# keys to exclude from dict in grid operations
+DICT_EXCLUDE_KEYS = {TYPE_TAG_STR, "attrs"}
```

Comment:
Why was this necessary here and might it be necessary in other places too (in the future)?

Reply:
Basically, `attrs` gets written into the `dict()` and the `grid` uses `self.dict(exclude={"type"})` to get the xyz items. If we don't exclude the `attrs`, it will get included in the dictionary and will break the grid.

Options moving forward are to keep adding to this, or try to change the grid iteration to just iterate through `'xyz'` directly. The latter sounds a bit more robust so I'll try that.

Reply:
yea this worked better, I just changed the `grid` code to iterate through `xyz` and that way if we change the contents of `dict()` we don't need to worry about this.